### PR TITLE
feature/[PATRO23-37]-Add Budget module

### PR DESCRIPTION
**Pull request by PiotrekkP**

### CODE REVIEW

**Comment by PiotrekkP** [comment]
Now I am wondering if branch should be named based on story or subtask from this story which in this case is "Create budget module". My bet is that I hurried up and named it wrong, because there are more people working on this story so branches names need to be different. But I'll wait for your comments.

**Comment by mateuszmuszynski** [comment]
> Now I am wondering if branch should be named based on story or subtask from this story which in this case is "Create budget module". My bet is that I hurried up and named it wrong, because there are more people working on this story so branches names need to be different. But I'll wait for your comments.

Yes, let's use subtask id for branch names. But let's leave this case as it is.

File: src/modules/budget/infrastructure/Migrations/20230313153024_InitialMigration.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+namespace Intive.Patronage2023.Modules.Budget.Infrastructure.Migrations
```
Comment (mateuszmuszynski) [inline src/modules/budget/infrastructure/Migrations/20230313153024_InitialMigration.cs:5]: Not sure whether we should copy migrations and entity builder. Maybe we can just remove this and DbSet from context.
@mateuszgiedzinski What do you think?

**Review by mateuszmuszynski — CHANGES_REQUESTED** [review]

File: src/modules/budget/infrastructure/Migrations/20230313153024_InitialMigration.cs (hunk 1 above)
Comment (mateuszgiedzinski) [reply src/modules/budget/infrastructure/Migrations/20230313153024_InitialMigration.cs:5]: I agree. We need to exclude Migrations directory from copy.

**Review by AlanP22 — APPROVED** [review]

**Review by a-lipert — APPROVED** [review]

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]
Add init migration and adjust readme with module migration script.

File: src/modules/budget/api/Properties/launchSettings.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
+﻿{
+  "$schema": "https://json.schemastore.org/launchsettings.json",
```
[the remark this answers is not in the document]
Comment (mateuszgiedzinski) [reply src/modules/budget/api/Properties/launchSettings.json:2]: still not removed

[the remark this answers is not in the document]
Comment (mateuszgiedzinski) [reply src/modules/budget/api/Properties/launchSettings.json]: Please remove whole file. We don't want to run module independently so launch settings is not  necessery.

File: src/modules/budget/api/Intive.Patronage2023.Modules.Budget.Api.csproj [hunk 3]
```diff
@@ -31,4 +31,8 @@
     <ProjectReference Include="..\application\Intive.Patronage2023.Modules.Budget.Application.csproj" />
     <ProjectReference Include="..\infrastructure\Intive.Patronage2023.Modules.Budget.Infrastructure.csproj" />
   </ItemGroup>
+
+  <ItemGroup>
```
Comment (mateuszgiedzinski) [inline src/modules/budget/api/Intive.Patronage2023.Modules.Budget.Api.csproj:35]: Properties directory is not needed. Do not leave empty directories.

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

**Review by mateuszgiedzinski — APPROVED** [review]

File: src/modules/budget/infrastructure/Data/BudgetDbContext.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+/// <summary>
+/// Database context.
+/// </summary>
+public class BudgetDbContext : DbContext
+{
+	/// <summary>
+	/// Initializes a new instance of the <see cref="BudgetDbContext"/> class.
+	/// </summary>
+	/// <param name="options">DbContext options.</param>
+	public BudgetDbContext(DbContextOptions<BudgetDbContext> options)
```
Comment (mateuszgiedzinski) [inline src/modules/budget/infrastructure/Data/BudgetDbContext.cs:17]: Add initial migration. 
Every change in db context or entities should have relevant migration.

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

**Pull request closed without merging**